bulla-contracts/src/mappings/BullaFactoringv2.ts

```diff
@@ -0,0 +1,297 @@
+import {
+  Deposit,
+  DepositMadeWithAttachment,
+  InvoiceFunded,
+  InvoiceImpaired,
+  InvoiceKickbackAmountSent,
+  InvoicePaid,
+  InvoicePaid__Params,
+  InvoiceUnfactored,
+  SharesRedeemedWithAttachment,
+  Withdraw
+} from "../../generated/BullaFactoringv2/BullaFactoringv2";
+import { getClaim } from "../functions/BullaClaimERC721";
+import {
+  createDepositMadeEvent,
+  createInvoiceFundedEvent,
+  createInvoiceImpairedEvent,
+  createInvoiceKickbackAmountSentEvent,
+  createInvoicePaidEvent,
+  createInvoiceUnfactoredEvent,
+  createSharesRedeemedEvent,
+  getDepositMadeEventId,
+  getSharesRedeemedEventId
+} from "../functions/BullaFactoringv2";
+import {
+  getIPFSHash_depositWithAttachment_v2,
+  getIPFSHash_redeemWithAttachment_v2,
+  getLatestPrice,
+  getOrCreateHistoricalFactoringStatistics,
+  getOrCreatePoolProfitAndLoss,
+  getOrCreatePricePerShare,
+  getOrCreateUser
+} from "../functions/common";
+import { DepositMadeEvent, SharesRedeemedEvent } from "../../generated/schema";
+
+export function handleInvoiceFunded(event: InvoiceFunded): void {
+  const ev = event.params;
```

Comment:
I think a more sustainable approach is you define all of these functions with a `version` parameter, so the core logic is not replicated.

Then, you define two functions, oen for V1 and one for V2.

example:

```
function handleInvoiceFunded(version: "v1" | "v2", event: InvoiceFunded) {...}`

export function handleInvoiceFunded_v1( event: InvoiceFunded)) {return handleInvoiceEvent("v1", event);}
export function handleInvoiceFunded_v2( event: InvoiceFunded)) {return handleInvoiceEvent("v2", event);}

```